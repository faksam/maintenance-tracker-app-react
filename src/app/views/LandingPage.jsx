import React from 'react';

import AuthForms from '../components/auth/AuthForms';

import creatRequest from '../../images/create-request.png';
import awaitApproval from '../../images/await-approval.png';
import trackProgress from '../../images/track-progress.png';
import getNotified from '../../images/get-notified.png';

const LandingPage = (props) => {
    const { history } = props;
    return (<div className="main-container">
        <div className="row">
            <h1 className="center header-logo">M - T - A</h1>
        </div>
        <div className="container">
            <div className="row center">
                <div className="col col-1-2">
                    <div className="row wall center">
                    </div>
                    <h3>M-T-A (Maintenance - Tracker - App) is an online maintenance service assistant that allows you to create
                    a maintenance request and track the progress of your request until it is resolved and completely
done.</h3>
                </div>
                <div className="col col-1-2 float-right">
                    <AuthForms history={history} />
                </div>
            </div>
        </div>
        <div className="container">
            <div className="row">
                <h3 className="center">HOW IT WORKS</h3>
                <hr />
            </div>
            <div className="row center">
                <div className="col col-1-4">
                    <div className="row">
                        <img className="description-img" src={creatRequest} />
                    </div>
                    <div className="row">
                        <h3>CREATE A REQUEST</h3>
                    </div>
                </div>
                <div className="col col-1-4">
                    <div className="row">
                        <img className="description-img" src={awaitApproval} />
                    </div>
                    <div className="row">
                        <h3>AWAIT APPROVAL</h3>
                    </div>
                </div>
                <div className="col col-1-4">
                    <div className="row">
                        <img className="description-img" src={trackProgress} />
                    </div>
                    <div className="row">
                        <h3>TRACK PROGRESS</h3>
                    </div>
                </div>
                <div className="col col-1-4">
                    <div className="row">
                        <img className="description-img" src={getNotified} />
                    </div>
                    <div className="row">
                        <h3>GET NOTIFIED</h3>
                    </div>
                </div>
            </div>
            <div className="row center">
                <h3>
                    With M-T-A there are no worries of leaving your office to look for the maitenance guy or waiting for days to know the progress
                    of your maitenance request.
</h3>
            </div>
            <br />
        </div>

        <div className="footer">
            <div className="row">
                <p>
                    Â© 2018 Copyright M-T-A by
<a target="_blank" href="http://www.fakunlesamuel.com">FakSam</a>
                </p>
            </div>
        </div>
    </div>
    );
};

export default LandingPage;
