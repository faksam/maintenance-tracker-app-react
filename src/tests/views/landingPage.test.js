import React from 'react';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import LandingPage from '../../app/views/LandingPage';

const props = {
  fullname: 'fakunlesamuel@gmail.com',
  email: 'fakunlesamuel@gmail.com',
  password: 'Lifeisarace',
  confirmPassword: 'Lifeisarace',
  handleSignIn: jest.fn(),
  handleSignUp: jest.fn(),
  handleChange: jest.fn(),
  viewClickedRequest: jest.fn(),
  handleCreateRequest: jest.fn(),
  hideModalRequest: jest.fn(),
  signoutUserAction: jest.fn(),
  loadRequestsAction: jest.fn(),
  request: [{}],
  history: {},
  user: {
    username: 'joeeasy',
    email: 'joeeasy@gmail.com',
    role: 'Admin',
    id: '6bf54873-6033-4ab0-855d-336a34e1dfbc',
    iat: 1540985284931
  },
};
const initialState = {
  auth: {
    isAuthenticated: true,
    error: {},
    user: {
      username: 'joeeasy',
      email: 'joeeasy@gmail.com',
      role: 'Admin',
      id: '6bf54873-6033-4ab0-855d-336a34e1dfbc',
      iat: 1540985284931
    }
  },
  requestReducer: {
    requests: []
  }

};
const mockStore = configureMockStore([thunk]);
const store = mockStore(initialState);

describe('SignInTabForm Component', () => {
  it('should render without throwing an error', () => {
    const wrapper = shallow(<LandingPage {...props} />);
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
  it('should set state to props', () => {
    const wrapper = shallow(<LandingPage {...props} />);
    expect(wrapper.exists()).toBe(true);
  });
});
