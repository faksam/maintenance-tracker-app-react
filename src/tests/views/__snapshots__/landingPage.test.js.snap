// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should render without throwing an error 1`] = `
<div
  className="main-container"
>
  <div
    className="row"
  >
    <h1
      className="center header-logo"
    >
      M - T - A
    </h1>
  </div>
  <div
    className="container"
  >
    <div
      className="row center"
    >
      <div
        className="col col-1-2"
      >
        <div
          className="row wall center"
        />
        <h3>
          M-T-A (Maintenance - Tracker - App) is an online maintenance service assistant that allows you to create a maintenance request and track the progress of your request until it is resolved and completely done.
        </h3>
      </div>
      <div
        className="col col-1-2 float-right"
      >
        <Connect(AuthForm)
          history={Object {}}
        />
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <h3
        className="center"
      >
        HOW IT WORKS
      </h3>
      <hr />
    </div>
    <div
      className="row center"
    >
      <div
        className="col col-1-4"
      >
        <div
          className="row"
        >
          <img
            className="description-img"
            src={Object {}}
          />
        </div>
        <div
          className="row"
        >
          <h3>
            CREATE A REQUEST
          </h3>
        </div>
      </div>
      <div
        className="col col-1-4"
      >
        <div
          className="row"
        >
          <img
            className="description-img"
            src={Object {}}
          />
        </div>
        <div
          className="row"
        >
          <h3>
            AWAIT APPROVAL
          </h3>
        </div>
      </div>
      <div
        className="col col-1-4"
      >
        <div
          className="row"
        >
          <img
            className="description-img"
            src={Object {}}
          />
        </div>
        <div
          className="row"
        >
          <h3>
            TRACK PROGRESS
          </h3>
        </div>
      </div>
      <div
        className="col col-1-4"
      >
        <div
          className="row"
        >
          <img
            className="description-img"
            src={Object {}}
          />
        </div>
        <div
          className="row"
        >
          <h3>
            GET NOTIFIED
          </h3>
        </div>
      </div>
    </div>
    <div
      className="row center"
    >
      <h3>
        With M-T-A there are no worries of leaving your office to look for the maitenance guy or waiting for days to know the progress of your maitenance request.
      </h3>
    </div>
    <br />
  </div>
  <div
    className="footer"
  >
    <div
      className="row"
    >
      <p>
        Â© 2018 Copyright M-T-A by
        <a
          href="http://www.fakunlesamuel.com"
          target="_blank"
        >
          FakSam
        </a>
      </p>
    </div>
  </div>
</div>
`;
